pages/options.js

```diff
@@ -0,0 +1,251 @@
+var p = photonui;
+
+var layout;
+var filterList;
+
+$(document).ready(function() {
+  layout = buildLayout();
+  
+  chrome.storage.local.get('filter-list', init);
+  
+  p.domInsert(layout.filterListBox, document.getElementById("filterList"));
+  
+  p.domInsert(layout.detailBox, document.getElementById("details"));
+});
+
+function buildLayout() {
+  var layout = {};
+  
+  layout.filterListBox = new p.BoxLayout({
+    orientation: "vertical",
+  });
+  
+  layout.detailBox = new p.GridLayout();
+  
+  return layout;
+}
+
+function init(storageResults) {
+  filterList = storageResults["filter-list"] ? storageResults["filter-list"] : [];
+  
+  buildFilterList();
+}
+
+function buildFilterList(activeIndex) {
+  var currentActiveId = $(".listButton.active").attr("id");
+  layout.filterListBox.empty();
+  
+  for(var i = 0; i < filterList.length; i++) {
+    var item = filterList[i];
+    
+    var listItem = new p.BoxLayout({
+      orientation: "horizontal",
+    });
+    
+    var listButton = new p.Button({
+      text: item.src,
+      tooltip: "Edit This Filter",
+      name: "listButton-"+i
+    });
+    
+    listButton.id = "listButton-"+i,
+    listButton.addClass("listButton");
+    listButton.registerCallback("clicked", "click", function(widget) {
+      buildDetails(filterList[+widget.name.split("-")[1]]);
+      
+      for(var i = 0; i < layout.filterListBox.children.length; i++) {
+        $(".listButton").removeClass("active");
```

Comment:
Why does this need to be executed `children.length` times? Doesn't `$(".listButton")` work globally, i.e., not context-sensitive, so it wouldn't care how many times it gets called? If I'm reading this right the same line gets executed `children.length` times, with no difference between them.